src/algebraic_topology/simplex_category.lean

```diff
@@ -402,4 +402,167 @@ full_subcategory_inclusion _
+            exact h a_lt_x },
+          simp [b_geq_x] }}⟩ with def_chi_2,
+  -- The two auxiliary functions equalize f
+  have f_comp_chi_i : f ≫ chi_1 = f ≫ chi_2,
+  { dsimp,
+    ext,
+    rw [hom.to_preorder_hom_mk, hom.to_preorder_hom_mk, preorder_hom.comp_coe],
+    rw [hom.to_preorder_hom_mk, hom.to_preorder_hom_mk, preorder_hom.comp_coe],
+    rw [function.comp_app, function.comp_app],
+    have f_x_1_ne_x : ¬((f.to_preorder_hom) x_1 = x) := h_ab x_1,
+    by_cases ((f.to_preorder_hom) x_1) ≤ x,
+    { have f_x_1_lt_x : ((f.to_preorder_hom) x_1) < x,
+      { cases nat.eq_or_lt_of_le h,
+        exfalso,
+        replace h_1 := subtype.eq h_1,
+        exact f_x_1_ne_x h_1,
+        exact h_1 },
+      simp [f_x_1_lt_x, h] },
+    { have n_f_x_1_lt_x : ¬(((f.to_preorder_hom) x_1) < x),
+    { erw not_iff_not_of_iff nat.lt_iff_le_not_le,
+      rw not_and,
+      intro h3, exfalso,
+      refine h h3 },
+      simp [n_f_x_1_lt_x, h] }},
+  resetI,
+  -- We now just have to show the two auxiliary functions are not equal.
+  rw category_theory.cancel_epi f at f_comp_chi_i, rename f_comp_chi_i eq_chi_i,
+  apply_fun hom.to_preorder_hom at eq_chi_i,
+  apply_fun preorder_hom.to_fun at eq_chi_i,
+  replace eq_chi_i := congr_fun eq_chi_i x,
+  dsimp at eq_chi_i,
+  have chi_1_x : (hom.to_preorder_hom chi_1) x = 0 := by { simp },
+  have chi_2_x : (hom.to_preorder_hom chi_2) x = 1 := by { simp },
+  rw [chi_1_x, chi_2_x] at eq_chi_i,
+  refine nat.zero_ne_one (fin.veq_of_eq eq_chi_i) },
+  { intro hyp_surj,
+    refine ⟨ by { intros l g h h_eq_comp,
+      dsimp at *,
+      ext,
+      set y := Exists.some (hyp_surj x) with d_y,
+      have y_eq_f_x : x = f.to_preorder_hom y,
+       { rw d_y,
+        exact (Exists.some_spec (hyp_surj x)).symm },
+      apply_fun hom.to_preorder_hom at h_eq_comp,
+      apply_fun preorder_hom.to_fun at h_eq_comp,
+      replace h_eq_comp := congr_fun h_eq_comp y,
+      dsimp at h_eq_comp,
+      rw [hom.to_preorder_hom_mk, preorder_hom.comp_coe, function.comp_app] at h_eq_comp,
+      rw [hom.to_preorder_hom_mk, preorder_hom.comp_coe, function.comp_app] at h_eq_comp,
+      rw ←y_eq_f_x at h_eq_comp,
+      refine fin.veq_of_eq h_eq_comp }⟩}
+end
+
+/-- A monomorphism in `simplex_category` must increase lengths-/
+lemma mono_le_length {x y : simplex_category} {f : x ⟶ y} :
+  (category_theory.mono f) → (x.len ≤ y.len) :=
+begin
+  intro hyp_f_mono,
+  have f_inj : function.injective f.to_preorder_hom.to_fun,
+  { exact mono_iff_injective.elim_left (hyp_f_mono) },
+  have card_leq := fintype.card_le_of_injective f.to_preorder_hom.to_fun f_inj,
+  simp at card_leq,
+  exact card_leq,
+end
+
+lemma mono_le_card {n m : ℕ} {f : [n] ⟶ [m]} : (category_theory.mono f) → (n ≤ m) :=
+by { refine @mono_le_length [n] [m] f}
+
+/-- An epimorphism in `simplex_category` must decrease lengths-/
+lemma epi_ge_length {x y : simplex_category} {f : x ⟶ y} :
+  (category_theory.epi f) → (x.len ≥ y.len) :=
```

Comment:
mathlib statements use `≤` rather than `≥`, so you should state this as `epi f → y.len ≤ x.len`